src/commands/cloudShell.ts

```diff
@@ -0,0 +1,124 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { IActionContext, parseError } from '@microsoft/vscode-azext-utils';
+import { exec } from 'child_process';
+import * as vscode from 'vscode';
+import { ext } from '../extensionVariables';
+
+// This command is meant to be called once vscode.dev connects to VS Code server running in Cloud Shell via Dev Tunnel.
+// As long as we're connected, it periodically does the following:
+//  1. refreshes the dev tunnel access token when connected for the VS Code Server
+//  2. calls the Cloud Shell /keepAlive endpoint to ensure Cloud Shell stays active
+//  3. calls the Cloud Shell /size endpoint to ensure the terminal stays active
+
+function delay(ms: number) {
+    return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+interface MaintainCloudShellConnectionOptions {
+    consoleUri: string;
+    terminalId: string;
+}
+
+export async function maintainCloudShellConnection(_context: IActionContext, options: MaintainCloudShellConnectionOptions) {
+    while (true) {
+        try {
+            await refreshDevTunnelAccessToken();
+        } catch (e) {
+            ext.outputChannel.error(parseError(e).message);
+        }
+
+        try {
+            await cloudShellKeepAlive(options.consoleUri);
+        } catch (e) {
+            ext.outputChannel.error(parseError(e).message);
+        }
+
+        try {
+            await cloudShellSize(options.consoleUri, options.terminalId);
+        } catch (e) {
+            ext.outputChannel.error(parseError(e).message);
+        }
+
+        await delay(1000 * 60 * 10 /* 10 minutes */);
```

Comment:
Is it documented how long it takes for the connection to die?

Reply:
No, the Cloud Shell engineers say "about every 10 minutes should work"